File: Code coach problems/Medium/C++/Convert_US_date_to_EU_date.cpp
#include <iostream>
#include <string>
using namespace std;
int main()
{
	int year=0, month=0, day=0;
	string date;
	cin >> date;
	if(!isdigit(date[0]))
	{
		if(date == "January")
			month = 1;
		else if(date  == "February")
			month = 2;
		else if(date  == "March")
			month = 3;
		else if(date  == "April")
			month = 4;
		else if(date  == "May")
			month = 5;
		else if(date  == "June")
			month = 6;
		else if(date  == "July")
			month = 7;
		else if(date  == "August")
			month = 8;
		else if(date  == "October")
			month = 9;
		else if(date  == "November")
			month = 10;
		else if(date  == "December")
			month = 11;
		else
			month = 12;
		cin >> date;
		int i = 0;
		while(date[i] != ',')
		{
			day = day*10 + date[i] - '0';
			i++;	
		}
		cin >> date;
		i = 0;
		while(i != date.size())
		{
			year = year*10 + date[i] - '0';
			i++;
		}
	}
	else
	{
		int i = 0;
		while(date[i] != '/')
		{
			month = month*10 + date[i] - '0';
			i++;
		}
		i++;
		
		while(date[i] != '/')
		{
			day = day*10 + date[i] - '0';
			i++;
		}
		i++;
		while(i != date.size())
		{
			year = year*10 + date[i] - '0';
			i++;
		}
	}
	
	cout << day << "/" << month << "/" << year;
	
	
}

File: Code coach problems/Medium/Python/Convert_US_date_to_EU_date.py
year = ""
month = ""
day = ""
date = input()
if not(date[0].isdigit()):
	i = 0
	while date[i] != ' ':
		i+=1
	buf = date[:i]
	if(buf == "January"):
		month = '1';
	elif(buf == "February"):
		month = '2';
	elif(buf == "March"):
		month = '3';
	elif(buf == "April"):
		month = '4';
	elif(buf == "May"):
		month = '5';
	elif(buf == "June"):
		month = '6';
	elif(buf == "July"):
		month = '7';
	elif(buf == "August"):
		month = '8';
	elif(buf == "October"):
		month = '9';
	elif(buf == "November"):
		month = '10';
	elif(buf == "December"):
		month = '11';
	else:
		month = '12';

	i+=1
	while date[i] != ',':
		day += date[i]
		i+=1
	i+=2
	while i != len(date):
		year += date[i]
	i+=1
else:
	i = 0
	while date[i] != '/':
		month += date[i]
		i+=1
	i+=1
	while date[i] != '/':
		day += date[i]
		i+=1
	i+=1
	while i != len(date):
		year += date[i]
		i+=1

print(day + "/" + month + "/" + year)

